go/interpreter/lfvm/instructions_test.go

```diff
@@ -1146,3 +1146,107 @@ func TestGenericCreate_MaxInitCodeSizeIsNotCheckedBeforeShanghai(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckSizeOFfsetUintOverflow_ReturnsAsExpected(t *testing.T) {
+
+	zero := uint256.NewInt(0)
+	one := uint256.NewInt(1)
+	u64overflow := new(uint256.Int).Add(uint256.NewInt(2), uint256.NewInt(math.MaxUint64))
+	// checkSizeOffsetUint64Overflow(offset, size) - parameter order
+
+	// size zero ignores offset
+	if want, got := error(nil), checkSizeOffsetUint64Overflow(u64overflow, zero); want != got {
+		t.Errorf("unexpected status after call, wanted %v, got %v", want, got)
+	}
+	// size overflow is reported
+	if want, got := errOverflow, checkSizeOffsetUint64Overflow(zero, u64overflow); want != got {
+		t.Errorf("unexpected status after call, wanted %v, got %v", want, got)
+	}
+	// offset overflow is reported
+	if want, got := errOverflow, checkSizeOffsetUint64Overflow(u64overflow, one); want != got {
+		t.Errorf("unexpected status after call, wanted %v, got %v", want, got)
+	}
+	// no overflow
+	if want, got := error(nil), checkSizeOffsetUint64Overflow(one, one); want != got {
+		t.Errorf("unexpected status after call, wanted %v, got %v", want, got)
+	}
+	// size+offset overflow is reported
+	if want, got := errOverflow, checkSizeOffsetUint64Overflow(uint256.NewInt(math.MaxUint64-1), uint256.NewInt(2)); want != got {
+		t.Errorf("unexpected status after call, wanted %v, got %v", want, got)
+	}
+
+}
+
+func TestGenericCall_ProperlyReportsErrors(t *testing.T) {
+
+	zero := uint256.NewInt(0)
+	one := uint256.NewInt(1)
+	u64overflow := new(uint256.Int).Add(uint256.NewInt(2), uint256.NewInt(math.MaxUint64))
+
+	tests := map[string]struct {
+		// stack order
+		// retSize, retOffset, inSize, inOffset, value, address, provided_gas
+		stack         []*uint256.Int
+		gas           tosca.Gas
+		expectedError error
+	}{
+		"input offset overflow": {
+			// size needs to be one, otherwise offset is ignored.
+			stack:         []*uint256.Int{zero, zero, one, u64overflow, zero, zero, zero},
+			expectedError: errOverflow,
+		},
+		"return Size overflow": {
+			stack:         []*uint256.Int{u64overflow, zero, zero, zero, zero, zero, zero},
+			expectedError: errOverflow,
+		},
+		"input memory too big": {
+			stack:         []*uint256.Int{zero, zero, uint256.NewInt(maxMemoryExpansionSize + 1), zero, zero, zero, zero},
+			expectedError: errMaxMemoryExpansionSize,
+		},
+		"not enough gas for output memory expansion": {
```

Comment:
Should there be a similar test for the input memory expansion?

Reply:
since `getSlice` has already been tested I did not want to make an exhaustive list here as well, but rather check that errors can be caught on both calls to `getSlice`, checking one error for input and another for output. If you think all cases should be duplicated for both input/output, I can add it as well.

Reply:
There are four different error branches, all should be reached by at least one of those cases: line 1044, 1048, 1054, and 1058.

Reply:
with the test cases as they are at this point, all 4 of those lines are reached.